File: tutorials/code/simp/cpp/esm/esm.cpp
#include <linkpred.hpp>
#include <iostream>
using namespace LinkPred::Simp;
int main() {
	int k = 10;
	// Create a prtedictor object
	Predictor p;
	// Load network from file
	p.loadnet("Zakarays_Karate_Club.edges");
	// Predict top k scores using an encoder-classifier predictor with Node2Vec as encoder and L2 similarity
	auto esv = p.predTopESM(k, "N2V", "L2");
	// Print scores
	std::cout << "N2V-L2\n";
	for (auto it = esv.begin(); it != esv.end(); ++it) {
		std::cout << it->i << "\t" << it->j << "\t" << it->score << std::endl;
	}
	// Predict top k scores using an encoder-similarity measure predictor with LINE as encoder and cosine similarity
	esv = p.predTopESM(k, "LIN", "CSM");
	// Print scores
	std::cout << "LIN-CSM\n";
	for (auto it = esv.begin(); it != esv.end(); ++it) {
		std::cout << it->i << "\t" << it->j << "\t" << it->score << std::endl;
	}
	return 0;
}

File: tutorials/code/simp/python/esm/esm.py
# Import the module
import LinkPredPython as lpp
k = 10;
# Create a predictor object
p = lpp.Predictor();
# Load network from file
p.loadnet("Zakarays_Karate_Club.edges");
# Predict top k scores using an encoder-classifier predictor with Node2Vec as encoder and L2 similarity
esv = p.predTopESM(k, "N2V", "L2");
# Print the scores
print("N2V-L2");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));
# Predict top k scores using an encoder-similarity measure predictor with LINE as encoder and cosine similarity
esv = p.predTopESM(k, "LIN", "CSM");
# Print the scores
print("LIN-CSM");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));
